docs/src/guide/self_consistent_field.jl

```diff
@@ -123,25 +123,25 @@
 
 using DFTK
 using LinearAlgebra
-function fixed_point_iteration(F, ρ₀, maxiter; tol)
+
+function fixed_point_iteration(F, ρ0, info0; maxiter)
     ## F:        The SCF step function
-    ## ρ₀:       The initial guess density
+    ## ρ0:       The initial guess density
+    ## info0:    The initial metadata
     ## maxiter:  The maximal number of iterations to be performed
-    ## tol:      The selected convergence tolerance
 
-    ρ  = ρ₀
-    Fρ = F(ρ)
+    ρ = ρ0
+    info = info0
     for n = 1:maxiter
-        ## If change less than tolerance, break iterations:
-        if norm(Fρ - ρ) < tol
+        Fρ, info = F(ρ, info)
+        if info.converged
             break
         end
-        ρ  = Fρ
-        Fρ = F(ρ)
+        ρ = Fρ
     end
```

Comment:
I agree this would be the "right" way to modify this function, but I'm wondering whether for demonstration purposes (this is supposed to be a low-entrance barrier example) it would be better not to "hide away" the convergence check. Don't you think @niklasschmitz ?

Reply:
Yes, that sounds better for pedagogical purposes. I will just add a note to link to the "custom solvers" example, where it should be explained how `info.converged` works.